charts/pulsar/templates/function-worker/_function_worker.tpl

```diff
@@ -0,0 +1,152 @@
+{{/*Define function worker service account*/}}
+{{- define "pulsar.function.serviceAccount" -}}
+{{- if .Values.functions.serviceAccount.create -}}
+    {{- if .Values.functions.serviceAccount.name -}}
+{{ .Values.functions.serviceAccount.name }}
+    {{- else -}}
+{{ template "pulsar.fullname" . }}-{{ .Values.functions.component }}-acct
+    {{- end -}}
+{{- else -}}
+{{ .Values.functions.serviceAccount.name }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Pulsar Broker Service URL
+*/}}
+{{- define "pulsar.function.broker.service.url" -}}
+{{- if not .Values.functions.useDedicateRunner -}}
+pulsar://localhost:6650
+{{- else -}}
+pulsar://{{ template "pulsar.fullname" . }}-{{ .Values.proxy.component }}:{{ .Values.proxy.ports.pulsar }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Pulsar Web Service URL
+*/}}
+{{- define "pulsar.function.web.service.url" -}}
+{{- if not .Values.functions.useDedicateRunner -}}
+http://localhost:8080
+{{- else -}}
+http://{{ template "pulsar.fullname" . }}-{{ .Values.proxy.component }}:{{ .Values.proxy.ports.http }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Pulsar Broker Service URL TLS
+*/}}
+{{- define "pulsar.function.broker.service.url.tls" -}}
+{{- if not .Values.functions.useDedicateRunner -}}
+pulsar+ssl://localhost:6651
+{{- else -}}
+pulsar+ssl://{{ template "pulsar.fullname" . }}-{{ .Values.proxy.component }}:{{ .Values.proxy.ports.pulsarssl }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Pulsar Web Service URL TLS
+*/}}
+{{- define "pulsar.function.web.service.url.tls" -}}
+{{- if not .Values.functions.useDedicateRunner -}}
+https://localhost:8443
+{{- else -}}
+https://{{ template "pulsar.fullname" . }}-{{ .Values.proxy.component }}:{{ .Values.proxy.ports.https }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Define function tls certs mounts
+*/}}
+{{- define "pulsar.function.certs.volumeMounts" -}}
+{{- if and .Values.tls.enabled .Values.tls.functions.enabled }}
+- name: function-certs
+  mountPath: "/pulsar/certs/function"
+  readOnly: true
+- name: ca
+  mountPath: "/pulsar/certs/ca"
+  readOnly: true
+{{- end }}
+{{- end }}
+
+{{/*
+Define function tls certs volumes
+*/}}
+{{- define "pulsar.function.certs.volumes" -}}
+{{- if and .Values.tls.enabled .Values.tls.broker.enabled }}
+- name: broker-certs
+  secret:
+    secretName: "{{ .Release.Name }}-{{ .Values.tls.functions.cert_name }}"
+    items:
+    - key: tls.crt
+      path: tls.crt
+    - key: tls.key
+      path: tls.key
+- name: ca
+  secret:
+    secretName: "{{ .Release.Name }}-ca-tls"
+    items:
+    - key: ca.crt
+      path: ca.crt
+{{- end }}
+{{- end }}
+
+{{/*
+Define the pulsar broker full service name
+*/}}
+{{- define "pulsar.function.service" -}}
+{{ template "pulsar.fullname" . }}-{{ .Values.functions.component }}
+{{- end }}
+
+
+{{/*
+Define function token mounts
+*/}}
+{{- define "pulsar.function.token.volumeMounts" -}}
+{{- if .Values.auth.authentication.enabled }}
+{{- if eq .Values.auth.authentication.provider "jwt" }}
+{{- if not .Values.auth.vault.enabled }}
```

Comment:
This chart does not support vault authentication, and it seems that you can remove this check